pkg/restapi/rp/operation/operations.go

```diff
@@ -17,32 +31,139 @@ type Handler interface {
 	Handle() http.HandlerFunc
 }
 
+type httpClient interface {
+	Post(url, contentType string, body io.Reader) (resp *http.Response, err error)
+}
+
 // Operation defines handlers
 type Operation struct {
-	handlers     []Handler
-	vcHTML       string
-	vcServiceURL string
+	handlers []Handler
+	vcHTML   string
+	vcsURL   string
+	client   httpClient
 }
 
 // Config defines configuration for rp operations
 type Config struct {
-	VCHTML       string
-	VCServiceURL string
+	VCHTML string
+	VCSURL string
+}
+
+// verifyCredentialResponse describes verify credential response
+type verifyCredentialResponse struct {
+	Verified bool   `json:"verified"`
+	Message  string `json:"message"`
+}
+
+// vc struct used to return vc data to html
+type vc struct {
+	Data string `json:"data"`
 }
 
 // New returns rp operation instance
 func New(config *Config) *Operation {
 	svc := &Operation{
-		vcHTML: config.VCHTML, vcServiceURL: config.VCServiceURL}
+		vcHTML: config.VCHTML, vcsURL: config.VCSURL, client: &http.Client{}}
 	svc.registerHandler()
 
 	return svc
 }
 
+// verifyVC
+func (c *Operation) verifyVC(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		c.writeErrorResponse(w, http.StatusInternalServerError,
+			fmt.Sprintf("failed to parse form: %s", err.Error()))
+
+		return
+	}
+
+	respData, err := c.httpPost(
+		c.vcsURL+"/verify", httpContentTypeJSON, []byte(r.Form.Get("vcDataInput")))
+	if err != nil {
+		c.writeErrorResponse(w, http.StatusBadRequest,
+			fmt.Sprintf("failed to verify vc: %s", err.Error()))
+
+		return
+	}
+
+	response := verifyCredentialResponse{}
+	if errUnmarshal := json.Unmarshal([]byte(respData), &response); errUnmarshal != nil {
+		c.writeErrorResponse(w, http.StatusInternalServerError,
+			fmt.Sprintf("failed to unmarshal vc: %s", errUnmarshal.Error()))
+
+		return
+	}
+
+	if !response.Verified {
+		c.writeErrorResponse(w, http.StatusBadRequest,
+			fmt.Sprintf("failed to verify vc: %s", response.Message))
+
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+
+	t, err := template.ParseFiles(c.vcHTML)
+	if err != nil {
+		c.writeErrorResponse(w, http.StatusInternalServerError,
+			fmt.Sprintf("unable to load html: %s", err.Error()))
+
+		return
+	}
+
+	if err := t.Execute(w, vc{Data: r.Form.Get("vcDataInput")}); err != nil {
+		log.Error(fmt.Sprintf("failed execute html template: %s", err.Error()))
+	}
+}
+
+func (c *Operation) httpPost(url, commContentType string, data []byte) (string, error) {
+	resp, err := c.client.Post(url, commContentType, bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+
+	if resp != nil {
```

Comment:
should do the resp == nil case in the block.

Reply:
https://github.com/trustbloc/edge-sandbox/pull/35/files#diff-141aceae11185429c1572fa974caf361R150